Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -39,14 +39,20 @@ function Get-AzureUri
 function Get-ProxyUri
 {
     param([String] [Parameter(Mandatory=$true)] $serverUrl)
+    
+    $proxyUri = [System.Uri]($env:AGENT_PROXYURL)
+    Write-Verbose -Verbose ("Reading proxy from the AGENT_PROXYURL environment variable. Proxy url specified={0}" -f $proxyUri.OriginalString)
 
-    $proxyUri = [Uri]$null
-    $proxy = [System.Net.WebRequest]::GetSystemWebProxy()
-
-    if ($proxy)
+    if($proxyUri -eq $null)
     {
-        $proxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
+        $proxy = [System.Net.WebRequest]::GetSystemWebProxy()
         $proxyUri = $proxy.GetProxy("$serverUrl")
+        Write-Verbose -Verbose ("Reading proxy from IE. Proxy url specified={0}" -f $proxyUri.OriginalString)
+    }
+
+    if($serverUrl -like "*$($proxyUri.Host)*")
```

Comment:
Can you get the host component for serverUrl similar to proxy and just compare the host instead of wildcard matching.